pytorch_lightning/trainer/training_loop.py

```diff
@@ -686,6 +684,7 @@ def run_training_batch(self, batch, batch_idx, dataloader_idx):
         splits = self._tbptt_split_batch(batch)
 
         for split_idx, split_batch in enumerate(splits):
+            self.split_idx = split_idx
```

Comment:
Do you think it's better to have this here instead of in `run_train_split_start`?

Reply:
Yes, two reasons 1) because this is where the loop starts so if we are tracking split_idx as instance variable, this is the place where we update it 2) we do the same for batch_idx one loop level above.